media_utils/video_puppeteer.py

```diff
@@ -187,12 +198,18 @@ def playback_done(video):
     # either still playing or there is a problem, so this will never
     # return True.
 
+    active_re = re.compile('NextGetSampleIndex:([\d]+)')
+
     debug_lines = video.get_debug_lines()
     if debug_lines is not None:
+        for occurence in active_re.findall(' '.join(debug_lines)):
+            if occurence.group(1) == '0':
+                raise VideoException('Video %s is stalled' % video)
```

Comment:
Aside from checking the buffered ranges, then, how exactly does one detect a stall? That's what this code is supposed to be doing.